src/main/java/edu/harvard/iq/dataverse/authorization/AuthenticationServiceBean.java

```diff
@@ -334,18 +353,16 @@ public AuthenticatedUser authenticate( String authenticationProviderId, Authenti
                 user = userService.updateLastLogin(user);
             }
             
-            /**
-             * @todo Why does a method called "authenticate" have the potential
-             * to call "createAuthenticatedUser"? Isn't the creation of a user a
-             * different action than authenticating?
```

Comment:
@michbarsinai I am very happy to see that you renamed this method. Thank you!

Reply:
You welcome :-)
I later actually recalled why I called it `authenticate` rather than `getCreate....`. The reason (which I think is still valid but I don't feel too strong about it) is encapsulation/design by contract and hiding. The basic idea was that after this call is done, you get an authenticated user, and you should not care whether it involved another row in the database - all you need to now is that it matches the passed login request and that it is now persisted. In that sense, the whole "get/create" is a bit "too much information".
I don't feel that strongly about this, but it's worth thinking about.